### DISCUSSION TITLE
Correct config to allow top-level await (typescript)

### DISCUSSION THREAD

**Discussion by tslocke** [adonisjs/core#3094] [Help]
(Apologies if this is not technically an Adonis question)

Typescript allows top-level awaits, if you configure (tsconfig.json):

```
  "compilerOptions": {
    "target": "esnext",
    "module": "esnext",
    ...
  }
```

However this leads to an error on startup `Cannot use import statement outside a module`, with the hint `To load an ES module, set "type": "module" in the package.json`

Setting type to module in package.json gives another error

`TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension "" for myproject/ace`

Is there a way to configure things to allow top-level await?

Thanks

**Comment by thetutlage** [comment]
Yes, AdonisJS does not work with ES modules as of now. It requires a lot of internal changes. Moving from CJS to ESM unfortunately is not that simple
